src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -30,11 +30,11 @@ public static Process[] GetProcessesByName(string processName, string machineNam
             var processes = new List<Process>();
             foreach (int pid in ProcessManager.EnumerateProcessIds())
             {
-                Interop.procfs.ParsedStat parsedStat;
-                if (Interop.procfs.TryReadStatFile(pid, out parsedStat, reusableReader) &&
+                if (Interop.procfs.TryReadStatFile(pid, out Interop.procfs.ParsedStat parsedStat, reusableReader) &&
+                    Interop.procfs.TryReadStatusFile(pid, out Interop.procfs.ParsedStatus parsedStatus, reusableReader) &&
```

Comment:
This call should be swapped with the string.Equals below it.  We should not pay the cost of parsing this second file if the process name from the first doesn't match.

Reply:
Done.